src/tasks/account/taskaccount.cpp

```diff
@@ -25,8 +25,15 @@ void TaskAccount::run() {
   connect(
       request, &NetworkRequest::requestFailed, this,
       [this](QNetworkReply::NetworkError error, const QByteArray&) {
-        logger.error() << "Account request failed" << error;
-        ErrorHandler::instance()->errorHandle(ErrorHandler::toErrorType(error));
+        logger.error() << "Account request failed" << error
+                       << "error-propagation:"
+                       << (m_errorPropagation == ErrorHandler::PropagateError
+                               ? "yes"
+                               : "no");
+        if (m_errorPropagation == ErrorHandler::PropagateError) {
```

Comment:
I am not a big fan of re-declaring this conditional in each task, it makes it easier for the meaning or "ErrorPropagation" to be changed in sutil ways as new tasks are implemented or changed. What if we declare a function on the Task super class like so:

```cpp
void handleTaskError(Error error) {
   if (m_errorPropagation == ErrorHandler::PropagateError) {
       ErrorHandler::instance()->errorHandle(
              ErrorHandler::toErrorType(error));
   }
}
```